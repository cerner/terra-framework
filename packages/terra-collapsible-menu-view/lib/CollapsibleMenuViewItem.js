"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _bind = _interopRequireDefault(require("classnames/bind"));
var _terraButton = _interopRequireDefault(require("terra-button"));
var _terraButtonGroup = _interopRequireDefault(require("terra-button-group"));
var _CollapsibleMenu = _interopRequireDefault(require("./_CollapsibleMenu"));
var _CollapsibleMenuViewModule = _interopRequireDefault(require("./CollapsibleMenuView.module.scss"));
var _excluded = ["icon", "isIconOnly", "isReversed", "text", "isSelected", "isDisabled", "subMenuItems", "shouldCloseOnClick", "boundingRef", "menuWidth"];
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
var cx = _bind.default.bind(_CollapsibleMenuViewModule.default);
var propTypes = {
  /**
   * Sets the item's text
   */
  text: _propTypes.default.string.isRequired,
  /**
   * Bounding container for the menu, will use window if no value provided.
   */
  boundingRef: _propTypes.default.func,
  /**
  * An optional icon. Nested inline with the text when provided
  */
  icon: _propTypes.default.element,
  /**
   * Indicates if only the icon should display when item is face-up. (Text should still be given to be displayed when item is in the menu)
   */
  isIconOnly: _propTypes.default.bool,
  /**
  * Reverses the position of the icon and text
  */
  isReversed: _propTypes.default.bool,
  /**
   * Indicates if the item start as selected when placed in a button group
   */
  isSelected: _propTypes.default.bool,
  /**
   * Indicates if the item should be disabled.
   */
  isDisabled: _propTypes.default.bool,
  /**
   * Indicates that clicking on this item while displayed inside a menu should close the menu.
   */
  shouldCloseOnClick: _propTypes.default.bool,
  /**
   * List of Menu.Items to display in a submenu when this item is selected.
   */
  subMenuItems: _propTypes.default.arrayOf(_propTypes.default.element),
  /**
   * Callback function for when the item is clicked
   */
  onClick: _propTypes.default.func,
  /**
   * A string representation of the width in px, limited to:
   * 160, 240, 320, 640, 960, 1280, 1760, or auto
   */
  menuWidth: _propTypes.default.oneOf(['160', '240', '320', '640', '960', '1280', '1760', 'auto'])
};
var contextTypes = {
  isCollapsibleGroupItem: _propTypes.default.bool,
  isCollapsibleMenuItem: _propTypes.default.bool
};
var defaultProps = {
  isSelected: false,
  isReversed: false,
  shouldCloseOnClick: true,
  isIconOnly: false
};
var CollapsibleMenuViewItem = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(CollapsibleMenuViewItem, _React$Component);
  var _super = _createSuper(CollapsibleMenuViewItem);
  function CollapsibleMenuViewItem(props) {
    var _this;
    (0, _classCallCheck2.default)(this, CollapsibleMenuViewItem);
    _this = _super.call(this, props);
    _this.setButtonNode = _this.setButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.getButtonNode = _this.getButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }
  (0, _createClass2.default)(CollapsibleMenuViewItem, [{
    key: "setButtonNode",
    value: function setButtonNode(node) {
      this.buttonNode = node;
    }
  }, {
    key: "getButtonNode",
    value: function getButtonNode() {
      return this.buttonNode;
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
        icon = _this$props.icon,
        isIconOnly = _this$props.isIconOnly,
        isReversed = _this$props.isReversed,
        text = _this$props.text,
        isSelected = _this$props.isSelected,
        isDisabled = _this$props.isDisabled,
        subMenuItems = _this$props.subMenuItems,
        shouldCloseOnClick = _this$props.shouldCloseOnClick,
        boundingRef = _this$props.boundingRef,
        menuWidth = _this$props.menuWidth,
        customProps = (0, _objectWithoutProperties2.default)(_this$props, _excluded);
      var _this$context = this.context,
        isCollapsibleGroupItem = _this$context.isCollapsibleGroupItem,
        isCollapsibleMenuItem = _this$context.isCollapsibleMenuItem;
      var attributes = _objectSpread({}, customProps);
      var item;
      if (isCollapsibleMenuItem) {
        item = /*#__PURE__*/_react.default.createElement(_CollapsibleMenu.default.Item, (0, _extends2.default)({}, attributes, {
          text: text,
          isSelected: isSelected && isCollapsibleGroupItem,
          isDisabled: isDisabled,
          subMenuItems: subMenuItems
        }));
      } else if (isCollapsibleGroupItem) {
        item = /*#__PURE__*/_react.default.createElement(_terraButtonGroup.default.Button, (0, _extends2.default)({}, attributes, {
          icon: icon,
          text: text,
          isDisabled: isDisabled
        }));
      } else if (subMenuItems && subMenuItems.length > 0) {
        item = /*#__PURE__*/_react.default.createElement(_CollapsibleMenu.default, {
          contentWidth: menuWidth,
          boundingRef: boundingRef,
          button: /*#__PURE__*/_react.default.createElement(_terraButton.default, (0, _extends2.default)({}, attributes, {
            icon: icon,
            text: text,
            isReversed: isReversed,
            isDisabled: isDisabled,
            isIconOnly: isIconOnly
          }))
        }, subMenuItems);
      } else {
        item = /*#__PURE__*/_react.default.createElement("div", {
          className: cx('face-up-item')
        }, /*#__PURE__*/_react.default.createElement(_terraButton.default, (0, _extends2.default)({}, attributes, {
          icon: icon,
          text: text,
          isReversed: isReversed,
          isDisabled: isDisabled,
          isIconOnly: isIconOnly
        })));
      }
      return item;
    }
  }]);
  return CollapsibleMenuViewItem;
}(_react.default.Component);
CollapsibleMenuViewItem.propTypes = propTypes;
CollapsibleMenuViewItem.defaultProps = defaultProps;
CollapsibleMenuViewItem.contextTypes = contextTypes;
CollapsibleMenuViewItem.Opts = {
  widths: _CollapsibleMenu.default.Opts.widths
};
var _default = CollapsibleMenuViewItem;
exports.default = _default;