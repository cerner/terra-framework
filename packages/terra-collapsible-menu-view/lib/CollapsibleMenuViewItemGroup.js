"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _bind = _interopRequireDefault(require("classnames/bind"));
var _terraButtonGroup = _interopRequireDefault(require("terra-button-group"));
var _CollapsibleMenu = _interopRequireDefault(require("./_CollapsibleMenu"));
var _CollapsibleMenuViewModule = _interopRequireDefault(require("./CollapsibleMenuView.module.scss"));
var _excluded = ["children", "onChange", "selectedKeys"];
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
var cx = _bind.default.bind(_CollapsibleMenuViewModule.default);
var propTypes = {
  /**
   * Callback function that is called when the group's selection state changes
   */
  onChange: _propTypes.default.func,
  /**
   * CollapsibleMenuView.Items to be grouped together
   */
  children: _propTypes.default.node.isRequired,
  /**
   * A list of keys of the CollapsibleMenuView.Items that should be selected.
   */
  selectedKeys: _propTypes.default.arrayOf(_propTypes.default.string)
};
var defaultProps = {
  selectedKeys: []
};
var childContextTypes = {
  isCollapsibleGroupItem: _propTypes.default.bool
};
var contextTypes = {
  isCollapsibleMenuItem: _propTypes.default.bool
};
var CollapsibleMenuViewItemGroup = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(CollapsibleMenuViewItemGroup, _React$Component);
  var _super = _createSuper(CollapsibleMenuViewItemGroup);
  function CollapsibleMenuViewItemGroup(props) {
    var _this;
    (0, _classCallCheck2.default)(this, CollapsibleMenuViewItemGroup);
    _this = _super.call(this, props);
    _this.handleOnChange = _this.handleOnChange.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }
  (0, _createClass2.default)(CollapsibleMenuViewItemGroup, [{
    key: "getChildContext",
    value: function getChildContext() {
      return {
        isCollapsibleGroupItem: true
      };
    }
  }, {
    key: "handleOnChange",
    value: function handleOnChange(event, selectedIndex) {
      if (this.props.onChange) {
        var selectedKey = selectedIndex;
        _react.default.Children.forEach(this.props.children, function (child, index) {
          if (selectedIndex === index) {
            selectedKey = child.key;
          }
        });
        this.props.onChange(event, selectedKey);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
        children = _this$props.children,
        onChange = _this$props.onChange,
        selectedKeys = _this$props.selectedKeys,
        customProps = (0, _objectWithoutProperties2.default)(_this$props, _excluded);
      var isCollapsibleMenuItem = this.context.isCollapsibleMenuItem;
      if (isCollapsibleMenuItem && selectedKeys.length) {
        return /*#__PURE__*/_react.default.createElement("li", {
          role: "none"
        }, /*#__PURE__*/_react.default.createElement(_CollapsibleMenu.default.ItemGroup, (0, _extends2.default)({}, customProps, {
          onChange: this.handleOnChange
        }), children));
      }
      if (isCollapsibleMenuItem) {
        return /*#__PURE__*/_react.default.createElement("div", customProps, children);
      }
      var buttonGroupClassNames = cx(['face-up-item', customProps.className]);
      return /*#__PURE__*/_react.default.createElement(_terraButtonGroup.default, (0, _extends2.default)({}, customProps, {
        onChange: onChange,
        className: buttonGroupClassNames,
        selectedKeys: selectedKeys
      }), children);
    }
  }]);
  return CollapsibleMenuViewItemGroup;
}(_react.default.Component);
CollapsibleMenuViewItemGroup.propTypes = propTypes;
CollapsibleMenuViewItemGroup.defaultProps = defaultProps;
CollapsibleMenuViewItemGroup.childContextTypes = childContextTypes;
CollapsibleMenuViewItemGroup.contextTypes = contextTypes;
var _default = CollapsibleMenuViewItemGroup;
exports.default = _default;