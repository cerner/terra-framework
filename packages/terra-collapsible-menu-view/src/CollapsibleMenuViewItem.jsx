import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Button from 'terra-button';
import ButtonGroup from 'terra-button-group';
import Menu from './_CollapsibleMenu';
import styles from './CollapsibleMenuView.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Sets the item's text
   */
  text: PropTypes.string.isRequired,

  /**
   * Bounding container for the menu, will use window if no value provided.
   */
  boundingRef: PropTypes.func,

  /**
  * An optional icon. Nested inline with the text when provided
  */
  icon: PropTypes.element,

  /**
   * Indicates if only the icon should display when item is face-up. (Text should still be given to be displayed when item is in the menu)
   */
  isIconOnly: PropTypes.bool,

  /**
  * Reverses the position of the icon and text
  */
  isReversed: PropTypes.bool,

  /**
   * Indicates if the item start as selected when placed in a button group
   */
  isSelected: PropTypes.bool,

  /**
   * Indicates if the item should be disabled.
   */
  isDisabled: PropTypes.bool,

  /**
   * Indicates that clicking on this item while displayed inside a menu should close the menu.
   */
  shouldCloseOnClick: PropTypes.bool,

  /**
   * List of Menu.Items to display in a submenu when this item is selected.
   */
  subMenuItems: PropTypes.arrayOf(PropTypes.element),

  /**
   * Callback function for when the item is clicked
   */
  onClick: PropTypes.func,

  /**
   * A string representation of the width in px, limited to:
   * 160, 240, 320, 640, 960, 1280, 1760, or auto
   */
  menuWidth: PropTypes.oneOf(['160', '240', '320', '640', '960', '1280', '1760', 'auto']),

  /**
   * Sets the button variant. One of neutral, emphasis, ghost, de-emphasis, action or utility.
   */
  variant: PropTypes.oneOf([Button.Opts.Variants.NEUTRAL, Button.Opts.Variants.EMPHASIS, Button.Opts.Variants.GHOST, Button.Opts.Variants['DE-EMPHASIS'], Button.Opts.Variants.ACTION, Button.Opts.Variants.UTILITY]),
};

const contextTypes = {
  isCollapsibleGroupItem: PropTypes.bool,
  isCollapsibleMenuItem: PropTypes.bool,
};

const defaultProps = {
  isSelected: false,
  isReversed: false,
  shouldCloseOnClick: true,
  isIconOnly: false,
  variant: Button.Opts.Variants.NEUTRAL,
};

class CollapsibleMenuViewItem extends React.Component {
  constructor(props) {
    super(props);
    this.setButtonNode = this.setButtonNode.bind(this);
    this.getButtonNode = this.getButtonNode.bind(this);
  }

  setButtonNode(node) {
    this.buttonNode = node;
  }

  getButtonNode() {
    return this.buttonNode;
  }

  render() {
    const {
      icon,
      isIconOnly,
      isReversed,
      text,
      isSelected,
      isDisabled,
      subMenuItems,
      shouldCloseOnClick,
      boundingRef,
      menuWidth,
      variant,
      ...customProps
    } = this.props;

    const { isCollapsibleGroupItem, isCollapsibleMenuItem } = this.context;
    const attributes = { ...customProps };
    let item;

    if (isCollapsibleMenuItem) {
      item = (
        <Menu.Item
          {...attributes}
          text={text}
          isSelected={isSelected && isCollapsibleGroupItem}
          isDisabled={isDisabled}
          subMenuItems={subMenuItems}
        />
      );
    } else if (isCollapsibleGroupItem) {
      item = (
        <ButtonGroup.Button
          {...attributes}
          icon={icon}
          text={text}
          isDisabled={isDisabled}
        />
      );
    } else if (subMenuItems && subMenuItems.length > 0) {
      item = (
        <Menu
          contentWidth={menuWidth}
          boundingRef={boundingRef}
          button={(
            <Button
              {...attributes}
              icon={icon}
              text={text}
              isReversed={isReversed}
              isDisabled={isDisabled}
              isIconOnly={isIconOnly}
              variant={variant}
            />
          )}
        >
          {subMenuItems}
        </Menu>
      );
    } else {
      item = (
        <div className={cx('face-up-item')}>
          <Button
            {...attributes}
            icon={icon}
            text={text}
            isReversed={isReversed}
            isDisabled={isDisabled}
            isIconOnly={isIconOnly}
            variant={variant}
          />
        </div>
      );
    }

    return item;
  }
}

CollapsibleMenuViewItem.propTypes = propTypes;
CollapsibleMenuViewItem.defaultProps = defaultProps;
CollapsibleMenuViewItem.contextTypes = contextTypes;
CollapsibleMenuViewItem.Opts = {
  widths: Menu.Opts.widths,
};

export default CollapsibleMenuViewItem;
