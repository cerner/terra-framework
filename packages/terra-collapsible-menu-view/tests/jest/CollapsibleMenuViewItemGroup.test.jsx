import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import CollapsibleMenuViewItemGroup from '../../src/CollapsibleMenuViewItemGroup';
import CollapsibleMenuViewItem from '../../src/CollapsibleMenuViewItem';

describe('CollapsibleMenuViewItemGroup', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallowWithIntl(
      (
        <CollapsibleMenuViewItemGroup>
          <CollapsibleMenuViewItem text="Testing" />
          <CollapsibleMenuViewItem text="Testing" />
        </CollapsibleMenuViewItemGroup>
      ),
    );
    expect(wrapper).toMatchSnapshot();
  });

  // Props
  it('should merge custom props', () => {
    const wrapper = shallowWithIntl(
      (
        <CollapsibleMenuViewItemGroup className="Testing">
          <CollapsibleMenuViewItem text="Testing" />
          <CollapsibleMenuViewItem text="Testing" />
        </CollapsibleMenuViewItemGroup>
      ),
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a selectable button group', () => {
    const wrapper = shallowWithIntl(
      (
        <CollapsibleMenuViewItemGroup selectedKeys={['key1']}>
          <CollapsibleMenuViewItem text="Testing" key="key1" />
          <CollapsibleMenuViewItem text="Testing" key="key2" />
        </CollapsibleMenuViewItemGroup>
      ),
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('Collapsible Menu Context', () => {
    it('should render a menu item group when selectable', () => {
      const context = { isCollapsibleMenuItem: true };
      const wrapper = shallowWithIntl(
        (
          <CollapsibleMenuViewItemGroup selectedKeys={['key1']}>
            <CollapsibleMenuViewItem text="Testing" key="key1" />
            <CollapsibleMenuViewItem text="Testing" key="key2" />
          </CollapsibleMenuViewItemGroup>
        ), { context },
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should merge custom props', () => {
      const context = { isCollapsibleMenuItem: true };
      const wrapper = shallowWithIntl(
        (
          <CollapsibleMenuViewItemGroup className="Testing">
            <CollapsibleMenuViewItem text="Testing" />
            <CollapsibleMenuViewItem text="Testing" />
          </CollapsibleMenuViewItemGroup>
        ), { context },
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should merge custom props when selectable', () => {
      const context = { isCollapsibleMenuItem: true };
      const wrapper = shallowWithIntl(
        (
          <CollapsibleMenuViewItemGroup className="Testing" selectedKeys={['key1']}>
            <CollapsibleMenuViewItem text="Testing" key="key1" />
            <CollapsibleMenuViewItem text="Testing" key="key2" />
          </CollapsibleMenuViewItemGroup>
        ), { context },
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should render menu items when selectedKeys is not set', () => {
      const context = { isCollapsibleMenuItem: true };
      const wrapper = shallowWithIntl(
        (
          <CollapsibleMenuViewItemGroup>
            <CollapsibleMenuViewItem text="Testing" />
            <CollapsibleMenuViewItem text="Testing" />
          </CollapsibleMenuViewItemGroup>
        ), { context },
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
