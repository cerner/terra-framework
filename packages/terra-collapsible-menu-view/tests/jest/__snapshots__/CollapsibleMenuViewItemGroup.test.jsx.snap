// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleMenuViewItemGroup Collapsible Menu Context should merge custom props 1`] = `
<div
  className="Testing"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    shouldCloseOnClick={true}
    text="Testing"
  />
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    shouldCloseOnClick={true}
    text="Testing"
  />
</div>
`;

exports[`CollapsibleMenuViewItemGroup Collapsible Menu Context should merge custom props when selectable 1`] = `
<li
  role="none"
>
  <MenuItemGroup
    className="Testing"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    onChange={[Function]}
  >
    <CollapsibleMenuViewItem
      isIconOnly={false}
      isReversed={false}
      isSelected={false}
      key="key1"
      shouldCloseOnClick={true}
      text="Testing"
    />
    <CollapsibleMenuViewItem
      isIconOnly={false}
      isReversed={false}
      isSelected={false}
      key="key2"
      shouldCloseOnClick={true}
      text="Testing"
    />
  </MenuItemGroup>
</li>
`;

exports[`CollapsibleMenuViewItemGroup Collapsible Menu Context should render a menu item group when selectable 1`] = `
<li
  role="none"
>
  <MenuItemGroup
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    onChange={[Function]}
  >
    <CollapsibleMenuViewItem
      isIconOnly={false}
      isReversed={false}
      isSelected={false}
      key="key1"
      shouldCloseOnClick={true}
      text="Testing"
    />
    <CollapsibleMenuViewItem
      isIconOnly={false}
      isReversed={false}
      isSelected={false}
      key="key2"
      shouldCloseOnClick={true}
      text="Testing"
    />
  </MenuItemGroup>
</li>
`;

exports[`CollapsibleMenuViewItemGroup Collapsible Menu Context should render menu items when selectedKeys is not set 1`] = `
<div
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    shouldCloseOnClick={true}
    text="Testing"
  />
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    shouldCloseOnClick={true}
    text="Testing"
  />
</div>
`;

exports[`CollapsibleMenuViewItemGroup should merge custom props 1`] = `
<ButtonGroup
  className="face-up-item Testing"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isBlock={false}
  selectedKeys={Array []}
>
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    shouldCloseOnClick={true}
    text="Testing"
  />
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    shouldCloseOnClick={true}
    text="Testing"
  />
</ButtonGroup>
`;

exports[`CollapsibleMenuViewItemGroup should render a default component 1`] = `
<ButtonGroup
  className="face-up-item"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isBlock={false}
  selectedKeys={Array []}
>
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    shouldCloseOnClick={true}
    text="Testing"
  />
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    shouldCloseOnClick={true}
    text="Testing"
  />
</ButtonGroup>
`;

exports[`CollapsibleMenuViewItemGroup should render a selectable button group 1`] = `
<ButtonGroup
  className="face-up-item"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isBlock={false}
  selectedKeys={
    Array [
      "key1",
    ]
  }
>
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    key="key1"
    shouldCloseOnClick={true}
    text="Testing"
  />
  <CollapsibleMenuViewItem
    isIconOnly={false}
    isReversed={false}
    isSelected={false}
    key="key2"
    shouldCloseOnClick={true}
    text="Testing"
  />
</ButtonGroup>
`;
